src/IK/src/QPInverseKinematics.cpp

```diff
@@ -696,43 +704,78 @@ QPInverseKinematics::build(std::weak_ptr<const ParametersHandler::IParametersHan
         }
 
-        Eigen::VectorXd weight;
         if (priority == 1)
         {
-            if (!taskGroup->getParameter("weight", weight))
+            std::string weightProviderType = "ConstantWeightProvider";
+            if (!taskGroup->getParameter("weight_provider_type", weightProviderType))
+            {
+                log()->warn("{} Unable to get the parameter 'weight_provider_type' for the task "
+                            "in the group "
+                            "'{}'. The default one will be used. Default: '{}'.",
+                            logPrefix,
+                            taskGroupName,
+                            weightProviderType);
+            }
+
+            auto weightProvider = BipedalLocomotion::System::WeightProviderFactory::createInstance(
+                weightProviderType);
+
+            if (weightProvider == nullptr)
+            {
+                log()->error("{} The weight provider '{}' has not been registered.",
+                             logPrefix,
+                             weightProviderType);
+                return IntegrationBasedIKProblem();
+            }
+
+            if (!weightProvider->initialize(taskGroup))
             {
-                log()->error("{} Unable to get the parameter 'weight' for the task in the group "
+                log()->error("{} Unable to initialize the weight provider for the task in the "
+                             "group "
                              "'{}'.",
                              logPrefix,
                              taskGroupName);
-                return std::make_pair(BipedalLocomotion::System::VariablesHandler(), nullptr);
+                return IntegrationBasedIKProblem();
             }
-        }
 
-        if (!taskInstance->initialize(taskGroup))
-        {
-            log()->error("{} Unable to task named '{}'.", logPrefix, taskGroupName);
-            return std::make_pair(BipedalLocomotion::System::VariablesHandler(), nullptr);
-        }
+            // the weight is considered only if priority is equal to 1
+            if (!solver->addTask(taskInstance, taskGroupName, priority, weightProvider))
+            {
+                log()->error("{} Unable to add the task named '{}' in the solver.",
+                             logPrefix,
+                             taskGroupName);
+                return IntegrationBasedIKProblem();
+            }
 
-        // the weight is considered only if priority is equal to 1
-        if (!solver->addTask(taskInstance, taskGroupName, priority, weight))
+            weights[taskGroupName] = weightProvider;
+
+        } else
         {
-            log()->error("{} Unable to add the task named '{}' in the solver.",
-                         logPrefix,
-                         taskGroupName);
-            return std::make_pair(BipedalLocomotion::System::VariablesHandler(), nullptr);
+            // here the priority must be equal to 0
+            assert(priority == 0);
```

Comment:
I would suggest using a different error-handling system here. It is quite possible to fall here when putting a number different from 0 and 1 in the conf file (a common example of RTFM) . Crashing the application seems a bit excessive.

Reply:
Done cd1a8037990151f89133600b7b9d797427ea5072